lib/Email/MIME/Encode.pm

```diff
@@ -26,14 +25,14 @@ sub maybe_mime_encode_header {
       header_name_length => $header_name_length,
     });
   }
-
-  return _object_encode($val, $charset, $header_name_length, $Email::MIME::Header::header_to_class_map{$header})
-    if exists $Email::MIME::Header::header_to_class_map{$header};
-
   my $min_wrap_length = 78 - $header_name_length + 1;
 
   return $val
-    unless _needs_mime_encode($val) || $val =~ /[^\s]{$min_wrap_length,}/;
+    if !ref $val && !_needs_mime_encode_addr($val) && $val !~ /[^\s]{$min_wrap_length,}/;
```

Comment:
This is wrong. `_needs_mime_encode_addr` is function for checking if some part of address/mailbox of address-like header needs to be encoded. It is not for arbitrary header.